src/scripts/main.js

```diff
@@ -1,3 +1,189 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+const table = body.querySelector('table');
+const tHead = table.querySelector('thead');
+const tBody = table.querySelector('tbody');
+
+body.insertAdjacentHTML('beforeend', `
+  <form 
+    action="#"
+    method="get"
+    class="new-employee-form"
+  >
+    <label>Name:
+      <input name="name" type="text" data-qa="name">
+    </label>
+    <label>Position:
+      <input name="position" type="text" data-qa="position">
+    </label>
+    <label>Office:
+      <select name="office" data-qa="office">
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>Age:
+      <input
+        name="age"
+        type="number"
+        data-qa="age"
+        min=0
+      >
+    </label>
+    <label>Salary:
+      <input
+        name="salary"
+        type="number"
+        data-qa="salary"
+        min=0
+      >
+    </label>
+    <button type="submit">Save to table</button>
+  </form>
+`);
+
+function pushNotification(title, description, type) {
+  body.insertAdjacentHTML('beforeend', `
+    <div class="notification ${type}" data-qa="notification">
+      <h2 class="title">${title}</h2>
+      <p>${description}</p>
+  </div>
+  `);
+
+  setTimeout(() => {
+    body.removeChild(document.querySelector('.notification'));
+  }, 2000);
+}
+
+function checkFormData(key, value) {
+  if (!value) {
+    pushNotification('An error occured', 'Empty field value', 'error');
+
+    return false;
+  }
+
+  if (key === 'name' && value.length < 4) {
+    pushNotification('An error occured', 'Invalid name value', 'error');
+
+    return false;
+  }
+
+  if (key === 'age' && ((+value < 18 || +value > 90) || isNaN(+value))) {
+    pushNotification('An error occured', 'Invalid age value', 'error');
+
+    return false;
+  }
+
+  return true;
+}
+
+tHead.addEventListener('click', (e) => {
+  const index = e.target.cellIndex;
+  const isSorted = e.target.matches('.sorted');
```

Comment:
It is better to use CSS classes only for styling purposes. Use JS variable to toggle your sort order